t/unit/transport/test_redis.py

```diff
@@ -226,42 +444,97 @@ def _get_response_error(self):
 
     def _new_queue(self, queue, **kwargs):
         for pri in self.priority_steps:
-            self.client._new_queue(self._q_for_pri(queue, pri))
+            self.client._new_queue(self._queue_for_priority(queue, pri))
 
     def pipeline(self):
-        return Pipeline(Client())
+        return FakePipelineLite(FakeRedisClientLite())
+
 
+class FakeRedisKombuChannel(kombu_redis.Channel):
+    _fanout_queues = {}
+
+    def __init__(self, *args, **kwargs):
+        super(FakeRedisKombuChannel, self).__init__(*args, **kwargs)
+
+    def _get_client(self):
+        return FakeRedisClient
+
+    def _create_client(self, asynchronous=False):
+        global _fake_redis_client
+        if _fake_redis_client is None:
+            _fake_redis_client = self._get_client()()
+        return _fake_redis_client
+
+    @contextmanager
+    def conn_or_acquire(self, client=None):
+        if client:
+            yield client
+        else:
+            yield self._create_client()
 
-class Transport(redis.Transport):
-    Channel = Channel
+    def _get_pool(self, asynchronous=False):
+        params = self._connparams(asynchronous=asynchronous)
+        self.keyprefix_fanout = self.keyprefix_fanout.format(db=params['db'])
+        return FakeRedisConnectionPool(**params)
+
+    def _get_response_error(self):
+        return ResponseError
+
+    def _new_queue(self, queue, **kwargs):
+        for pri in self.priority_steps:
+            self.client._new_queue(self._queue_for_priority(queue, pri))
+
+    def pipeline(self):
+        yield _get_fake_redis_client().pipeline()
+
+    def basic_publish(self, message, exchange='', routing_key='', **kwargs):
+        self._inplace_augment_message(message, exchange, routing_key)
+        # anon exchange: routing_key is the destination queue
+        return self._put(routing_key, message, **kwargs)
+
+
+class FakeRedisKombuTransportLite(kombu_redis.Transport):
+    Channel = FakeRedisKombuChannelLite
+
+    def __init__(self, *args, **kwargs):
+        super(FakeRedisKombuTransportLite, self).__init__(*args, **kwargs)
 
     def _get_errors(self):
         return ((KeyError,), (IndexError,))
 
 
+class FakeRedisKombuTransport(FakeRedisKombuTransportLite):
+    Channel = FakeRedisKombuChannel
+
+
```

Comment:
So, without getting into too much detail -- the above is nearly 500 lines of adaptations and setting up clients on top of FakeRedis.
1) If you're already refactoring, I'd take these out to their own file. Maybe more than one.
2) Why is all this necessary? there seem to be no comments explaining it, and coming in knowing nothing, I'd basically expect a FakeRedis-based test-suite to just set up FakeRedis as the Redis server, and then run everything, as is, against that -- 10-20 lines of setup before the actual tests begin, not 500.

I note that about half of these lines are pre-existing, but IIUC that was before FakeRedis -- so the test basically included "a small FakeRedis" inside it. Now that an actual FakeRedis is used, I'd expect these to go away. Or, otherwise, explain the need for them.